native/src/main/scala/sqs4s/auth/Credentials.scala

```diff
@@ -62,32 +72,69 @@ object Credentials {
   /** Load Credentials in this order:
     * Environment variables
     * Java system properties
-    * Instance profile credentials– used on EC2 instances, and delivered through
+    * Instance profile credentials – used on EC2 instances, and delivered through
     * the Amazon EC2 metadata service.
     *
     * @param client        http client
     * @param ttl           session token's time to live
     * @param refreshBefore a small duration to refresh the token before it expires
+    * @param allowRedirect allow client to follow redirected URL
     * @tparam F an effect which represents the side effects
     * @return
     */
   def chain[F[_]: Concurrent: Timer](
     client: Client[F],
     ttl: FiniteDuration = 6.hours,
-    refreshBefore: FiniteDuration = 5.minutes
+    refreshBefore: FiniteDuration = 5.minutes,
+    allowRedirect: Boolean = true
   ): Resource[F, Credentials[F]] = {
     val env = envVar[F]
     val sys = sysProp[F]
-    val container = containerMetadata[F](client, ttl, refreshBefore)
-    val instance = instanceMetadata[F](client, ttl, refreshBefore)
+    val container =
+      containerMetadata[F](client, ttl, refreshBefore, allowRedirect)
+    val instance =
+      instanceMetadata[F](client, ttl, refreshBefore, allowRedirect)
 
-    val tryAllInOrder = env orElse sys orElse container orElse instance
+    val tryAllInOrder =
+      env orElse sys orElse container orElse instance
 
     tryAllInOrder.handleErrorWith { _ =>
       Resource.eval(NoValidAuthMethodError.raiseError)
     }
   }
 
+  /** Load Credentials in this order:
+    * Environment variables
+    * Java system properties
+    * Instance profile credentials – used on EC2 instances, and delivered through
+    * the Amazon EC2 metadata service
+    * STS assume role
```

Comment:
My suggestion is that STS is rolled into the `chain` method, after line 80.

Reply:
the reason I didn't roll STS inside `chain` is because STS reads a local file which requires a `Blocker` (a separate blocking threadpool) whereas others mechanism don't have that dependency.